library/core/src/ops/range.rs

```diff
@@ -147,6 +147,44 @@ impl<Idx: PartialOrd<Idx>> Range<Idx> {
     }
 }
 
+impl<Idx: core::cmp::Ord> Range<Idx> {
+    /// Compares the range to a scalar. Returns `Ordering::Equal`,
+    /// if the range contains the scalar. If not, returns `Ordering::Less`
+    /// or `Ordering::Greater`, depending on whether the range is
+    /// below or above the scalar.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// #![feature(cmp_scalar)]
+    /// # fn main() -> Result<(), usize> {
+    /// # struct File {
+    /// #    seqnum: u32,
+    /// #    range: core::ops::Range<usize>,
+    /// # }
+    /// let files = vec![
+    ///     File { seqnum: 0, range: 0..1000 },
+    ///     File { seqnum: 1, range: 1000..2200 },
+    ///     File { seqnum: 2, range: 2200..3900 },
+    ///     File { seqnum: 3, range: 3900..5000 },
+    /// ];
+    /// let target = 1600;
+    /// let index = files.binary_search_by(|f| f.range.cmp_scalar(target))?;
+    /// assert_eq!(files[index].seqnum, 1);
+    /// # Ok(()) }
+    /// ```
+    #[unstable(feature = "cmp_scalar", issue = "none")]
+    pub fn cmp_scalar(&self, scalar: Idx) -> core::cmp::Ordering {
+        if self.end <= scalar {
```

Comment:
Reminder that the `Range` might be degenerate.

What's this method supposed to return for something like `(100..0).cmp_scalar(50)`?  That's arguably both `Less` *and* `Greater`.

Reply:
Oh, you are absolutely correct, I didn't think of that case. Ugh, in that case it would make more sense to have `PartialOrd` -style semantics and return an `Option<Ordering>`...

Reply:
Made it to return `Option<Ordering>`. That makes the API bit more unergonomic, but I guess there's nothing we can do about that. (Expect that having a method that panics on degenerate ranges, but that sounds like a footgun. If the user is absolutely sure their ranges are always well-formed, they should call `.expect()` by themselves.)